src/components/Attachments/AttachmentCarousel/index.native.tsx

```diff
@@ -31,24 +33,28 @@ function AttachmentCarousel({report, reportActions, parentReportActions, source,
     useEffect(() => {
         const parentReportAction = report.parentReportActionID && parentReportActions ? parentReportActions[report.parentReportActionID] : undefined;
         const attachmentsFromReport = extractAttachmentsFromReport(parentReportAction, reportActions);
-
-        const initialPage = attachmentsFromReport.findIndex(compareImage);
+        let attachmentsFromNote: Attachment[] = [];
+        if (type === CONST.ATTACHMENT_TYPE.NOTE && accountID) {
+            attachmentsFromNote = extractAttachmentsFromNote(report.reportID, accountID);
+        }
+        const targetAttachments = type === CONST.ATTACHMENT_TYPE.REPORT ? attachmentsFromReport : attachmentsFromNote;
```

Comment:
Instead of parsing all report attachments upfront and conditionally deciding whether to use them, we can refactor the logic like this:

```ts
let attachments: Attachment[] = [];
if (type === CONST.ATTACHMENT_TYPE.NOTE && accountID) {
  attachments = extractAttachmentsFromNote(report.reportID, accountID);
} else {
  attachments = extractAttachmentsFromReport(parentReportAction, reportActions);
}
```

This way, we only parse the required attachments. If there's a possibility that `attachmentsFromReport` won't be used, we should avoid parsing them unnecessarily.